main.go

```diff
@@ -4,47 +4,136 @@ import (
 	"fmt"
 	"math/rand"
 	"time"
+	"sort"
 )
 
-type Carta struct {
-	Valore int
-	Seme   string
+type Card struct {
+	Value int
+	Seed   string
 }
 
-func New(v int, s string) *Carta {
-	return &Carta{Valore: v, Seme: s}
+type Deck struct {
+	Cards []*Card
 }
 
-func (c Carta) String() string {
-	if c.Valore < 11 {
-		return fmt.Sprintf("%s", c.Valore)
-	}
-
-	semi := map[int]string{1: "A", 11: "J", 12: "Q", 13: "K"}
-	return semi[c.Valore]
+//Costruttore per struttura Card
+func New(v int, s string) *Card {
+	return &Card{Value: v, Seed: s}
 }
 
-func CreaMazzo() [52]*Carta {
-	colori := []string{"Cuori", "Quadri", "Picche", "Fiori"}
-	var mazzo [52]*Carta
-	i := 0
-	for _, c := range colori {
-		for num := 1; num <= 13; num++ {
-			mazzo[i] = New(num, c)
-			i++
+//Costruttore struttura Deck
+func CreateDeck(start int) *Deck {
+	seeds := []string{"Cuori", "Quadri", "Picche", "Fiori"}
```

Comment:
Secondo me potresti dividere questa funzione in 2
1. Una che inizializza il mazzo
2. L'altra che mescola.

Cosi se trovi un modo per mescolare meglio, puoi semplicemente sostituire la funziona di mescolamento